### PR TITLE
[rpc] Add rate limit data to metrics

### PR DESCRIPTION
**Pull request by AlexiaChen**
Add message to metrics when rpc Access frequency exceeds limit via RPC

### CODE REVIEW

File: rpc/metrics.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+func init() {
+	prom.PromRegistry().MustRegister(
+		rpcRateLimitCounterVec,
+	)
+}
+
+var (
+	rpcRateLimitCounterVec = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "rpc",
```
Comment (LeoHChen) [inline rpc/metrics.go:17]: please use 'hmy' as the namespace for all harmony metrics. you can use "rpc" for subsystem.

**Review by LeoHChen — COMMENTED** [review]
did you check the metrics in local test?  can you print out the local test log by querying the metrics?

**Comment by LeoHChen** [comment]
use the following command to check your metrics
```bash
curl -s http://localhost:9900/metrics | grep ^hmy
```

File: rpc/metrics.go (hunk 1 above)
Comment (AlexiaChen) [reply rpc/metrics.go:17]: good point! Thanks

**Comment by AlexiaChen** [comment]
> 
> 
> use the following command to check your metrics
> 
> ```shell
> curl -s http://localhost:9900/metrics | grep ^hmy
> ```

Test it localloy：

```txt
mathxh@MathxH:~/go/src/github.com/harmony-one/harmony/bin$ curl -s http://localhost:9900/metrics | grep ^hmy
hmy_consensus_blskeys{index="0",pubkey="000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"} 0
hmy_consensus_finality_bucket{le="800"} 0
hmy_consensus_finality_bucket{le="1000"} 0
hmy_consensus_finality_bucket{le="1250"} 0
hmy_consensus_finality_bucket{le="1562.5"} 0
hmy_consensus_finality_bucket{le="1953.125"} 0
hmy_consensus_finality_bucket{le="2441.40625"} 0
hmy_consensus_finality_bucket{le="3051.7578125"} 0
hmy_consensus_finality_bucket{le="3814.697265625"} 0
hmy_consensus_finality_bucket{le="4768.37158203125"} 0
hmy_consensus_finality_bucket{le="5960.4644775390625"} 0
hmy_consensus_finality_bucket{le="+Inf"} 0
hmy_consensus_finality_sum 0
hmy_consensus_finality_count 0
hmy_downloader_num_blocks_inserted_beacon_helper 0
hmy_node_metadata{key="version",value="Harmony (C) 2020. harmony, version v7067-v4.0.0-134-g857d9463-dirty (mathxh@ 2021-06-04T14:38:10+0800)"} 1
hmy_rpc_rpc_over_ratelimit_trigger{rate_limit="300"} 18818
hmy_stream_bytes_read 0
hmy_stream_bytes_write 0
hmy_stream_msg_read 0
hmy_stream_msg_write 0
hmy_stream_num_server_request 0
hmy_stream_server_request_delay_bucket{le="0.01"} 0
hmy_stream_server_request_delay_bucket{le="0.02"} 0
hmy_stream_server_request_delay_bucket{le="0.04"} 0
hmy_stream_server_request_delay_bucket{le="0.08"} 0
hmy_stream_server_request_delay_bucket{le="0.16"} 0
hmy_stream_server_request_delay_bucket{le="+Inf"} 0
hmy_stream_server_request_delay_sum 0
hmy_stream_server_request_delay_count 0
```

File: rpc/metrics.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+		rpcRateLimitCounterVec,
+	)
+}
+
+var (
+	rpcRateLimitCounterVec = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "hmy",
+			Subsystem: "rpc",
+			Name:      "rpc_over_ratelimit_trigger",
```
Comment (LeoHChen) [inline rpc/metrics.go:19]: this additional rpc is redundant, please change it to `over_ratelimiter`

**Review by LeoHChen — APPROVED** [review]
two minor comments: 
* The current default rate limit of 300 seems too low, please change it to 1000 instead.  
* just shorten the name of the metrics

Approving it, please fix the above two minor comments.

**Review by rlan35 — APPROVED** [review]

**Pull request merged**